api/v1beta1/machine_types.go

```diff
@@ -61,6 +61,16 @@ const (
 	// search each annotation for during the pre-terminate.delete lifecycle hook
 	// to pause reconciliation of deletion. These hooks will prevent removal of
 	// an instance from an infrastructure provider until all are removed.
+	//
+	// Notes for Machines managed by KCP (starting with Cluster API v1.8.2):
+	// * KCP adds its own pre-terminate hook on all Machines it controls. This is done to ensure it can later remove
+	//   the etcd member right before Machine termination (i.e. before InfraMachine deletion). If the etcd member would be
+	//   removed earlier, kubelet would start failing and the Node drain would not work.
```

Comment:
```suggestion
	//   the etcd member right before Machine termination (i.e. before InfraMachine deletion). 
```
this it is true only for K8s>=1.31, and it is already described it in in the next point.

Same in the migration page